### PR TITLE
added edge-case testing

### PR DESCRIPTION
**Pull request by Manuel-Schmid**

### CODE REVIEW

File: backend/blog/tests/test_categories.py [hunk 1]
```diff
@@ -22,21 +22,57 @@ def test_create_category(client_query):
...
     )
 
     content = json.loads(response.content)
-    assert (content is not None)
-    assert (content['data'] is not None)
-    assert (content['data']['createCategory'] is not None)
-    assert (content['data']['createCategory']['category'] is not None)
+    assert content is not None
+    assert content['data'] is not None
+    data_create_category = content['data']['createCategory']
```
Comment (sytch) [inline backend/blog/tests/test_categories.py:35]: hier hast du auch das problem, dass es einen keyerror geben könnte wenn es `createCategory` nicht gibt.
wie du es elegant machen kannst wäre so:
```
data = content.get('data', None)  # wenn der key nicht vorhanden ist, wird data auf None gesetzt
assert data is not None
data_create_category = data.get('createCategory', None)
assert data_create_category is not None
...
```

File: backend/blog/tests/test_categories.py [hunk 2]
```diff
@@ -22,21 +22,57 @@ def test_create_category(client_query):
...
+    assert data_create_category is not None
+    assert data_create_category['category'] is not None
+    assert data_create_category['success'] == True
 
-    category = content['data']['createCategory']['category']
-    assert (category is not None)
-    assert (category['name'] == 'test_category4')
+    category = data_create_category['category']
+    assert category is not None
+    assert category['name'] == 'test_category4'
```
Comment (sytch) [inline backend/blog/tests/test_categories.py:42]: dito keyerror

File: backend/blog/tests/test_posts.py [hunk 3]
```diff
@@ -87,10 +223,12 @@ def test_update_post(client_query, posts):
     content = json.loads(response.content)
     assert (content is not None)
     assert (content['data'] is not None)
-    assert (content['data']['updatePost'] is not None)
-    assert (content['data']['updatePost']['post'] is not None)
+    data_update_post = content['data']['updatePost']
+    assert (data_update_post is not None)
```
Comment (sytch) [inline backend/blog/tests/test_posts.py:164]: die klammern noch entfernen

File: backend/blog/tests/test_comments.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    content = json.loads(response.content)
+    assert content is not None
+    data = content.get('data', None)
+    assert data is not None
+    data_create_comment = data.get('createComment', None)
+    assert data_create_comment is not None
+    assert data_create_comment['success'] is True
+    data_comment = data_create_comment.get('comment', None)
+    assert data_comment['title'] == 'test'
```
Comment (sytch) [inline backend/blog/tests/test_comments.py:70]: hier müsste noch `assert data_comment is not None` rein
